tests/4.5/target_teams_distribute_parallel_for/test_target_teams_distribute_parallel_for_default.c

```diff
@@ -0,0 +1,83 @@
+//===----  test_target_teams_distribute_parallel_for_default.c- combined consutrct -===//
+// 
+// OpenMP API Version 4.5 Nov 2015
+// 
+// testing the combined construct target teams distribute parallel for
+//
+//===---------------------------------------------------------------------===//
+
+#include <omp.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include "ompvv.h"
+
+
+int DefaultFirstPrivate() {
+  int ErrCount = 0;
+  int Arr[32];
+  for (int i = 0; i < 32; ++i) {
+    Arr[i] = i;
+  }
+
+#pragma omp target teams distribute parallel for num_teams(2) thread_limit(10)\
+        default(firstprivate)
+  for (int i = 0; i < 32; ++i) {
+    if (Arr[i] != i) {
+#pragma omp atomic
+      ErrCount += 1;
+    }
+  }
+  return ErrCount;
+}
+
+int DefaultPrivate() {
+  int ErrCount = 0;
+  int CONST = 123;
+  int Arr[32];
+  for (int i = 0; i < 32; ++i) {
+    Arr[i] = i;
+  }
+
+#pragma omp target teams distribute parallel for num_teams(2) thread_limit(10)\
+        default(private)
+  for (int i = 0; i < 32; ++i) {
+    CONST += 10;
```

Comment:
since CONST is privatized, its value is undefined (could be garbage).

Reply:
Hi @spophale ,
I observed that, when we use private for any variable, Inside each threads its variable is '0'. I thought I can use this to test "default(private)". Should I avoid assuming the values would be zero ?

Reply:
OpenMP Version 4.5, Section 2.15.3.3 private Clause, page 194, line 6-7 say the following:

> The new list item is initialized, or has an undefined initial value, as if it had been locally declared without an initializer.

The C11 standard says that non-static objects at block scope declared without explicit initializers are left uninitialized.
Therefore, we should avoid assuming the values would be zero.

Reply:
@lthakur007 yes, don't assume value is zero.

Reply:
Hi @spophale ,
Thank you for pointing out. I did the required changes...